pkg/provisioningv2/rke2/planner/drain.go

```diff
@@ -85,5 +92,21 @@ func (p *Planner) undrain(entry *planEntry) (bool, error) {
 		return false, p.store.updatePlanSecretLabelsAndAnnotations(entry)
 	}
 
+	if err := checkForDrainError(entry, "undraining"); err != nil {
+		// This is the only place true and an error is returned to indicate that draining is ongoing, but there is an error.
+		return true, err
+	}
+
+	// The annotations will be removed when undrain is done
 	return entry.Metadata.Annotations[rke2.UnCordonAnnotation] == "", nil
 }
+
+func checkForDrainError(entry *planEntry, drainStep string) error {
```

Comment:
Can we make this a godoc comment to indicate the scope of this functionality is only for looking for transient connectivity errors?

Reply:
Yes, I will add a comment.

Reply:
I understand the functionality here. The name of the function is vague.